src/strands/tools/mcp/mcp_from_config.py

```diff
@@ -0,0 +1,228 @@
+    env: Dict[str, str] = field(default_factory=dict)
+    cwd: Optional[str] = None
+
+    def __post_init__(self) -> None:
+        """Set transport type after initialization."""
+        self.transport_type = MCPTransportType.STDIO
+
+    def create_transport_callable(self) -> Callable[[], MCPTransport]:
+        """Create STDIO transport callable."""
+
+        def transport_callable() -> MCPTransport:
+            server_params = StdioServerParameters(
+                command=self.command,
+                args=self.args,
+                env=self.env,
+                cwd=self.cwd,
+            )
+            return stdio_client(server_params)
+
+        return transport_callable
+
+
+@dataclass
+class HTTPTransportConfig(MCPTransportConfig):
+    """Configuration for HTTP transport."""
+
+    url: str = ""
+    authorization_token: Optional[str] = None
+    headers: Dict[str, str] = field(default_factory=dict)
+
+    def __post_init__(self) -> None:
+        """Set transport type after initialization."""
+        self.transport_type = MCPTransportType.STREAMABLE_HTTP
+
+    def create_transport_callable(self) -> Callable[[], MCPTransport]:
+        """Create STREAMABLE_HTTP transport callable."""
+
+        def transport_callable() -> MCPTransport:
+            headers = self.headers.copy()
+            if self.authorization_token:
+                headers["Authorization"] = f"Bearer {self.authorization_token}"
+            return streamablehttp_client(self.url, headers=headers if headers else None)
+
+        return transport_callable
+
+
+@dataclass
+class SSETransportConfig(MCPTransportConfig):
+    """Configuration for SSE transport."""
+
+    url: str = ""
+    authorization_token: Optional[str] = None
+    headers: Dict[str, str] = field(default_factory=dict)
+
+    def __post_init__(self) -> None:
+        """Set transport type after initialization."""
+        self.transport_type = MCPTransportType.SSE
+
+    def create_transport_callable(self) -> Callable[[], MCPTransport]:
+        """Create SSE transport callable."""
+
+        def transport_callable() -> MCPTransport:
+            headers = self.headers.copy() if self.headers else {}
+            if self.authorization_token:
+                headers["Authorization"] = f"Bearer {self.authorization_token}"
+            return sse_client(self.url, headers=headers if headers else None)
+
+        return transport_callable
+
+
+def _infer_transport_from_url(url: str) -> str:
+    """Infer transport type from URL when none specified.
+
+    - If path contains '/sse' (optionally followed by / ? & or end), treat as 'sse'
+    - Otherwise default to 'streamable-http'
+    """
+    try:
+        path = (urlparse(url).path or "").lower()
+    except Exception:
+        return "streamable-http"
+    return "sse" if re.search(r"/sse(/|\?|&|$)", path) else "streamable-http"
+
+
+class MCPServerConfig:
+    """Configuration for an MCP server following MCP standards."""
```

Comment:
Is this truly a standard or a De facto standard?  If the former, then let's link to the standard, if the latter let's document what the format is (both in code and in our docs) to make it obvious to readers/customers